src/main/java/testProject/Test2.java

```diff
@@ -0,0 +1,32 @@
+package testProject;
+
+import java.io.BufferedReader;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStreamReader;
+
+public class Test2 {
+	
+	public static void main(String[] args) throws IOException {
+		// TODO Auto-generated method stub
+		String someFile="jeff.txt";
+
+		FileInputStream localFileInputStream = null;
+		try {
+			localFileInputStream = new FileInputStream(someFile);
+		} catch (FileNotFoundException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+		}
+		InputStreamReader localInputStreamReader = new InputStreamReader(localFileInputStream);
+		BufferedReader reader = new BufferedReader(localInputStreamReader);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *reader*. Currently, there are execution paths that do not contain closure statements. Either a) close *reader* in a try-finally block or b) close the resource by declaring *reader* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).